sieves/tasks/predictive/ner/bridges.py

```diff
@@ -0,0 +1,524 @@
+            Examples:
+            ----------
+            {%- for example in examples %}
+                Text: "{{ example.text }}":
+                Entities: [
+                    {%- for entity in example.entities %}
+                    {
+                        "text": "{{ entity.text }}",
+                        
+                        "entity_type": "{{ entity.entity_type }}"
+                    }{%- if not loop.last %}, {% endif %}
+                    {%- endfor %}
+                ]
+            {% endfor -%}
+            ----------
+        {% endif %}
+
+        For each entity:
+        - Extract the exact text of the entity
+        - Include a SHORT context string that contains ONLY the entity and AT MOST 3 words before and 3 words after it. 
+          DO NOT include the entire text as context. DO NOT include words that are not present in the original text 
+          as introductory words (Eg. 'Text:' before context string).
+        - Specify which type of entity it is (must be one of the provided entity types)
+
+        IMPORTANT:
+        - If the same entity appears multiple times in the text, extract each occurrence separately with its own context
+
+        Text: {{ text }}
+        Entity Types: {{ entity_types }}
+        """
+
+    @property
+    def _prompt_signature_description(self) -> str | None:
+        return """
+        Extract named entities from the provided text. For each entity found:
+        - Extract the exact text of the entity
+        - Include a SHORT context string that contains ONLY the entity and AT MOST 3 words before and 3 words 
+          after it. DO NOT include the entire text as context.
+        - Specify the entity type from the provided list of entity types
+        Only extract entities of the specified types.
+        """
+
+    @cached_property
+    def _prompt_signature(self) -> type[pydantic.BaseModel]:
+        entity_types = self._entities
+        LiteralType = Literal[*entity_types]  # type: ignore[valid-type]
+
+        class EntityWithContext(pydantic.BaseModel):
+            text: str
+            context: str
+            entity_type: LiteralType
+
+        class Prediction(pydantic.BaseModel):
+            entities: list[EntityWithContext] = []
+
+        if self.prompt_signature_description:
+            Prediction.__doc__ = jinja2.Template(self.prompt_signature_description).render()
+
+        return Prediction
+
+    def consolidate(
+        self, results: Iterable[pydantic.BaseModel], docs_offsets: list[tuple[int, int]]
+    ) -> Iterable[pydantic.BaseModel]:
+        results = list(results)
+        prediction_class = self._prompt_signature  # Get the class, not an instance
+
+        # Process each document (which may consist of multiple chunks)
+        for doc_offset in docs_offsets:
+            doc_results = results[doc_offset[0] : doc_offset[1]]
+            # Skip if no results for this document
+            if not doc_results:
```

Comment:
This must never happen. Result may be empty or in the worst case `None`, depending on whether the prompt + result conversion worked, but `doc_results` is a slice of a list containing these results and should never be falsy.

Is this condition met when you test your code?